### PR TITLE
Updated the maximum open file limit section

### LINKED ISSUE #3338: macOS prerequisites: open files limit in 2.0.2+
The macOS prerequisites doc has a section "Increase the maximum open file limit" detailing how to set a system-wide higher open file limit for processes using `launchd`/`launchctl` or `ulimit`.

Since https://github.com/dotnet/coreclr/pull/14054 has been merged and will be part of the 2.0.2 release, this should no longer be necessary as this appears to have been a bug in CoreCLR trying to set a higher limit itself, but with a wrong value. The value should now be set automatically to the system's highest allowed value (~10500 IIRC) and no manual intervention is needed.

cc @mairaw

### PR DESCRIPTION
**Pull request by JRAlexander**
# Updated the maximum open file limit section in MacOS .NET Core Prerequisites 

Fixes #3338

### CODE REVIEW

File: docs/core/macos-prerequisites.md [hunk 1]
```diff
@@ -62,9 +62,9 @@ Download and install the .NET Core SDK from [.NET Downloads](https://www.microso
 
 ---
 
-## Increase the maximum open file limit
+## Increase the maximum open file limit (.NET Core versions before .NET Core 2.0.2) 
 
-The default open file limit on macOS may not be sufficient for some .NET Core workloads, such as restoring projects or running unit tests.
+In older .NET Core versions (before .NET Core 2.0.2), the default open file limit on macOS may not be sufficient for some .NET Core workloads, such as restoring projects or running unit tests.
```
Comment (BillWagner) [inline docs/core/macos-prerequisites.md]: Should you clarify if this is the SDK or the runtime version? (I believe SDK from the context.)

**Review by BillWagner — APPROVED** [review]
This LGTM.

I had one question, if you wanted to address it.

File: docs/core/macos-prerequisites.md (hunk 1 above)
Comment (JRAlexander) [reply docs/core/macos-prerequisites.md]: That's a great question. I believe you are correct, but I will clarify and update. Thanks, @BillWagner, your insight is always spot-on! 😄

**Comment by JRAlexander** [comment]
Closing and reopening for build retrigger

**Pull request merged**